lib/domain/core/value_object.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'either.dart';
+import 'errors.dart';
+import 'failures.dart';
+
+// part 'value_objects.freezed.dart';
```

Comment:
Usually, I use Freezed for simple Data types and Union types, as seen on bloc states and entities. Since I'm not using all the extra features Freezed provides us in the ValueObject class. I found it easier to use with the @immutable decorator (also simpler to inherit). It can always be changed if necessary. I would be happy to see more elegant solutions, always eager to learn more.